spec/graphql/dataloader/nonblocking_dataloader_spec.rb

```diff
@@ -0,0 +1,261 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+if Fiber.respond_to?(:scheduler) # Ruby 3+
+  describe GraphQL::Dataloader::NonblockingDataloader do
+    class NonblockingSchema < GraphQL::Schema
+      class SleepSource < GraphQL::Dataloader::Source
+        def fetch(keys)
+          max_sleep = keys.max
+          # t1 = Time.now
+          # puts "----- SleepSource => #{max_sleep} "
+          sleep(max_sleep)
+          # puts "----- SleepSource done #{max_sleep} after #{Time.now - t1}"
+          keys.map { |_k| max_sleep }
```

Comment:
is this producing an array with same size of keys and with max_sleep in all positions?

Following this approach might be more clear what's doing:
```ruby
 Array.new(keys.size, max_sleep)
 ```

(Not super clear to me what this array will be needed for)